packages/platform-server/src/profiler.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+// tslint:disable:ban
+
+const PERFORMANCE_MARK_PREFIX = '🅰️';
+
+export let runMethodAndMeasurePerf = noopRunMethodAndMeasurePerf;
+
+function runAndMeasurePerfImpl<T>(label: string, syncMethod: () => T): T;
+async function runAndMeasurePerfImpl<T>(label: string, asyncMethod: () => Promise<T>): Promise<T>;
+async function runAndMeasurePerfImpl<T>(
+  label: string,
+  asyncMethod: () => Promise<T> | T,
+): Promise<T> {
+  const labelName = `${PERFORMANCE_MARK_PREFIX}:${label}`;
+  const startLabel = `start:${labelName}`;
+  const endLabel = `end:${labelName}`;
+
+  try {
+    /* tslint:disable:ban */
+    performance.mark(startLabel);
+    /* tslint:enable:ban */
+
+    const returnValue = asyncMethod();
+    if (returnValue instanceof Promise) {
+      return await returnValue;
+    } else {
+      return returnValue;
+    }
+  } finally {
+    // will be executed before the returns
+    /* tslint:disable:ban */
+    performance.mark(endLabel);
+    performance.measure(labelName, startLabel, endLabel);
+    performance.clearMarks(startLabel);
+    performance.clearMarks(endLabel);
+    /* tslint:enable:ban */
+  }
+}
+
+function noopRunMethodAndMeasurePerf<T>(label: string, asyncMethod: () => Promise<T>): Promise<T> {
+  return asyncMethod();
+}
+
+let warningLoged = false;
+
+/**
+ * This enables an internal performance profiler for SSR apps
+ *
+ * It should not be imported in application code
+ */
+export function enableSsrProfiling() {
+  if (!warningLoged && (!performance.mark || !performance.measure)) {
```

Comment:
I'd propose adding a `typeof` check for extra safety:
```suggestion
  if (!warningLogged && (typeof performance === 'undefined' || !performance.mark || !performance.measure)) {
```